cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/FileBatchSource.java

```diff
@@ -0,0 +1,210 @@
+    "is reading in files with the File log naming convention of YYYY-MM-DD-HH-mm-SS-Tag. The TimeFilter " +
+    "reads in files from the previous hour if the timeTable field is left blank. So if it's currently " +
+    "2015-06-16-15 (June 16th 2015, 3pm), it will read in files that contain 2015-06-16-14 in the filename. " +
+    "If the field timeTable is present, then it will read files in that haven't been read yet.";
+  private static final String FILESYSTEM_PROPERTIES_DESCRIPTION = "JSON of the properties needed for the " +
+    "distributed file system. The formatting needs to be as follows:\n{\n\t\"<property name>\" : " +
+    "\"<property value>\", ...\n}. For example, the property names needed for S3 are \"fs.s3n.awsSecretAccessKey\" " +
+    "and \"fs.s3n.awsAccessKeyId\".";
+  private static final String PATH_DESCRIPTION = "Path to file(s) to be read. If a directory is specified, " +
+    "terminate the path name with a \'/\'";
+  private static final String TABLE_DESCRIPTION = "Name of the Table that keeps track of the last time files " +
+    "were read in.";
+  private static final String INPUT_FORMAT_CLASS_DESCRIPTION = "Name of the input format class, which must be a " +
+    "subclass of FileInputFormat. Defaults to TextInputFormat";
+  private static final String FILESYSTEM_DESCRIPTION = "Distributed file system to read in from.";
+  private static final Gson GSON = new Gson();
+  private static final Logger LOG = LoggerFactory.getLogger(FileBatchSource.class);
+  private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd-HH-mm");
+
+  private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+
+  private final FileBatchConfig config;
+  private KeyValueTable table;
+  private Date prevMinute;
+
+  public FileBatchSource(FileBatchConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    if (config.timeTable != null) {
+      pipelineConfigurer.createDataset(config.timeTable, KeyValueTable.class, DatasetProperties.EMPTY);
+    }
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+    //calculate date one minute ago, rounded down to the nearest minute
+    prevMinute = new Date(context.getLogicalStartTime() - TimeUnit.MINUTES.toMillis(1));
+    Calendar cal = Calendar.getInstance();
+    cal.setTime(prevMinute);
+    cal.set(Calendar.SECOND, 0);
+    cal.set(Calendar.MILLISECOND, 0);
+    prevMinute = cal.getTime();
+    System.out.print(FILESYSTEM_PROPERTIES_DESCRIPTION);
+    Job job = context.getHadoopJob();
+    if (config.fileSystemProperties != null) {
+      Map<String, String> properties = GSON.fromJson(config.fileSystemProperties, MAP_STRING_STRING_TYPE);
+      for (Map.Entry<String, String> entry : properties.entrySet()) {
+        job.getConfiguration().set(entry.getKey(), entry.getValue());
```

Comment:
nit: To make the code more readable, is it better to extract ``job.getConfiguration()`` out to a variable